doc/index.rst

```diff
@@ -40,6 +40,7 @@ Sections
    api/api.rst
    samples/samples.rst
    reference/kconfig/index.rst
+   tools/board-support-tools.rst
```

Comment:
I really do not like to see this added to the top-level index. We need to find a better place for this, especially because this is being referenced from boards and not significant on its own.